### PR TITLE
Feat/vm logger

### PR DESCRIPTION
**Pull request by MoonOld**
implements a much lighter logger than env_logger to avoid too much burden resulted by vm lib linking
now supports
* read log level from env
* log with timestamp

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -1839,9 +1820,9 @@ name = "vm"
 version = "0.1.0"
 dependencies = [
  "backtrace",
- "env_logger",
  "immix",
  "internal_macro",
+ "log",
  "rand 0.8.5",
 ]
```
[the remark this answers is not in the document]
Comment (MoonOld) [reply Cargo.lock:1828]: 自动生成的你懂个鸡毛

File: vm/src/gc/mod.rs [hunk 2]
```diff
@@ -10,14 +11,21 @@ mod _immix {
...
+    // fn immix_gc_init_deprecated(ptr: *mut u8) {
+    //     _ = env_logger::Builder::from_env(
+    //         env_logger::Env::new()
+    //             .filter_or("GC_LOG", "error")
+    //             .write_style("GC_LOG_COLOR"),
+    //     )
+    //     .try_init();
+    //     immix::gc_init(ptr)
+    // }
+
```
Comment (Chronostasys) [inline vm/src/gc/mod.rs:24]: 去了

File: vm/src/logger/mod.rs [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+};
+
+///
+/// SimpleLogger is a logger with almost nothing dependency exclude std and log crates
+pub struct SimpleLogger {
+    log_limit: LevelFilter,
+}
+
+impl SimpleLogger {
+    ///
```
Comment (Chronostasys) [inline vm/src/logger/mod.rs:15]: 为啥开头多加一行

File: vm/src/logger/mod.rs [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+        Self::init_default_max_level(SimpleLogger {
+            log_limit: match LevelFilter::from_str(Self::get_env_log_level(log_level_name).as_str())
+            {
+                Ok(level_filter) => level_filter,
+                Err(_) => default_level,
+            },
+        });
+    }
+
+    // fn init(logger:Self) {
```
Comment (Chronostasys) [inline vm/src/logger/mod.rs:37]: 这种都去掉

**Review by Chronostasys — CHANGES_REQUESTED** [review]

**Review by Chronostasys — CHANGES_REQUESTED** [review]

**Review by Chronostasys — APPROVED** [review]
LGTM!

**Pull request merged**